src/routes/jira/workspaces/jira-workspaces-get.test.ts

```diff
@@ -0,0 +1,235 @@
+	const response = {
+		success: true,
+		workspaces: [] as { id: string; name: string; canCreateContainer: boolean }[]
+	};
+
+	let limit: number;
+
+	if (subscriptions.length < DEFAULT_LIMIT) {
+		limit = subscriptions.length;
+	} else {
+		limit = DEFAULT_LIMIT;
+	}
+
+	for (let i = 0; i < limit; i++) {
+		const subscription = subscriptions[i];
+		const repository = repositories[i];
+		const workspace = {
+			id: subscription.id.toString(),
+			name: repository.repoOwner,
+			canCreateContainer: false
+		};
+
+		response.workspaces.push(workspace);
+	}
+
+	return response;
+};
+
+describe("Workspaces Get", () => {
+	let app: Application;
+	let installation: Installation;
+	let jwt: string;
+
+	beforeEach(async () => {
+		installation = await Installation.install({
+			host: jiraHost,
+			sharedSecret: "shared-secret",
+			clientKey: "jira-client-key"
+		});
+
+		jwt = encodeSymmetric({
+			qsh: "context-qsh",
+			iss: "jira-client-key"
+		}, await installation.decrypt("encryptedSharedSecret", getLogger("test")));
+	});
+
+	it("Should return a 400 status if no Subscription is found for host", async () => {
+		app = express();
+		app.use((req, _, next) => {
+			req.log = getLogger("test");
+			next();
+		});
+		app.use(getFrontendApp());
+
+		await supertest(app)
+			.get("/jira/workspaces/search?searchQuery=Atlas")
+			.query({
+				jwt
+			})
+			.expect(res => {
+				expect(res.status).toBe(400);
+				expect(res.text).toContain(Errors.MISSING_SUBSCRIPTION);
+			});
+	});
+
+	it("Should return 200 status and empty array if no connected workspaces are found", async () => {
+		app = express();
+		app.use((req, _, next) => {
+			req.log = getLogger("test");
+			next();
+		});
+		app.use(getFrontendApp());
+
+		await createSubscriptions(jiraHost, 1);
+
+		await supertest(app)
+			.get("/jira/workspaces/search?searchQuery=atlas")
+			.query({
+				jwt
+			})
+			.expect(res => {
+				expect(res.text).toContain(JSON.stringify([]));
+				expect(res.status).toBe(200);
+			});
+	});
+
+	it("Should return all orgs when no searchQuery param is provided", async () => {
+		app = express();
+		app.use((req, _, next) => {
+			req.log = getLogger("test");
+			next();
+		});
+		app.use(getFrontendApp());
+
+		const subscriptions = await createSubscriptions(jiraHost, 2);
+		const repositories = await createRepositories(subscriptions);
+		const response = generateMockResponse(subscriptions, repositories);
+
+		await supertest(app)
+			.get("/jira/workspaces/search")
+			.query({
+				jwt
+			})
+			.expect(res => {
+				expect(res.text).toContain(JSON.stringify(response));
+				expect(res.status).toBe(200);
+			});
+	});
+
+	it("Should return all orgs that match provided searchQuery", async () => {
+		app = express();
+		app.use((req, _, next) => {
+			req.log = getLogger("test");
+			next();
+		});
+		app.use(getFrontendApp());
+
+		const subscriptions = await createSubscriptions(jiraHost, 3);
+		await createRepos(subscriptions);
```

Comment:
I'm using `createRepositories` for tests where I don't care what the repoOwner name is (in the event no searchQuery is passed and we return all orgs). However, in this case, where I want more control over naming, I created a separate function so I can be more explicit.